iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectContext.java

```diff
@@ -19,24 +19,72 @@
  */
 package org.sonar.iac.kubernetes.visitors;
 
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.iac.kubernetes.model.ProjectResource;
+
+import static org.sonar.iac.helm.HelmFileSystem.retrieveHelmProjectFolder;
+
 /**
  * Data class to provide information about the project. This allows to share cross-file knowledge to the individual checks.
  */
 public final class ProjectContext {
+  private final Map<String, Map<String, Set<ProjectResource>>> projectResourcePerNamespacePerPath = new HashMap<>();
+  private final FileSystem fileSystem;
 
-  private ProjectContext() {
+  private ProjectContext(FileSystem fileSystem) {
+    this.fileSystem = fileSystem;
   }
 
-  public static Builder builder() {
-    return new Builder();
+  /**
+   * Get all resources of a given {@code clazz} in a given {@code namespace} and that are accessible to a file with the given {@code path}.
+   * This means that the resources can be in the same file, or in the same directory, or in the descendant directories, but not in the ancestor directories.
+   */
+  public Set<ProjectResource> getProjectResources(String namespace, String path, Class<? extends ProjectResource> clazz) {
+    if (projectResourcePerNamespacePerPath.containsKey(namespace)) {
+      var resourcesPerPath = projectResourcePerNamespacePerPath.get(namespace);
+
+      var filePath = Path.of(path);
+      var basePath = Optional.ofNullable(retrieveHelmProjectFolder(filePath, fileSystem))
+        .or(() -> Optional.ofNullable(filePath.getParent()))
+        .map(Path::normalize)
+        .map(Path::toUri)
+        .map(URI::toString)
+        .orElse("");
+
+      return resourcesPerPath.entrySet().stream()
+        .filter(entry -> basePath.isEmpty() || entry.getKey().startsWith(basePath))
```

Comment:
In which case is it `basePath.isEmpty()` true? Can you add a test case?

Reply:
Only if the file is located in the FS root. But then, basePath can be set to `/`. I'll try to add a test case